src/collector.rs

```diff
@@ -119,11 +120,11 @@ pub async fn init_future(config: Config) -> Result<(), Error> {
 }
 
 pub fn new(config: Config) -> (RecorderHandle, impl Future<Output = ()>) {
-    let (collect_sender, collect_receiver) = mpsc::channel(1024);
+    let (collect_sender, mut collect_receiver) = mpsc::channel(1024);
     let (emit_sender, emit_receiver) = mpsc::channel(1024);
     let mut message_stream = Box::pin(
         stream::select(
-            collect_receiver.map(Message::Datum),
+            stream::poll_fn(move |cx| collect_receiver.poll_recv(cx)).map(Message::Datum),
```

Comment:
Is the tokio mpsc channel better than the futures one? A bit clumsy that it doesn't implement StreamExt

Reply:
It should be a bit faster than the futures one, so since we already use tokio we may as well use it.